src/solvers/bicgstab_solver.hpp

```diff
@@ -34,18 +34,20 @@ namespace solvers {
 struct BiCGSTABParams {
   MGParams mg_params;
   int max_iters = 1000;
-  Real residual_tolerance = 1.e-12;
+  std::shared_ptr<Real> residual_tolerance = std::make_shared<Real>(1.e-12);
```

Comment:
Out of curiosity: Why is this shared pointer?

Reply:
So that it can be pulled out by a downstream code and updated within a task in a task list.